core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -392,15 +396,22 @@ public IndicesQueriesRegistry buildQueryParserRegistry() {
 
         for (QueryRegistration<?> query : queries) {
             QueryParser<? extends QueryBuilder<?>> parser = query.parser;
-            for (String name: query.names) {
+            for (String name: query.parseField.getAllNamesIncludedDeprecated()) {
```

Comment:
maybe, I would also like to simplify things, cause at the moment we internally need QueryRegistration as we delay the actual registration to when buildQueryParserRegistry gets called. I may try this once I migrate the other register method. Or give up if it gets too big as part of this PR.